internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+			}
+		}
+	}
+
+	// Scope list for comparing values in matchNameBtwnScopes variable
+	scopeNames := []string{}
+
+	// Test scope names for dynamic true, string must be prepended with *
+	if internaltypes.IsDefined(model.Scopes) {
+		scopeElems := model.Scopes.Elements()
+		for _, scopeElem := range scopeElems {
+			scopeElemObjectAttrs := scopeElem.(types.Object)
+			scopeEntryName := scopeElemObjectAttrs.Attributes()["name"].(basetypes.StringValue).ValueString()
+			scopeNames = append(scopeNames, scopeEntryName)
+			scopeEntryIsDynamic := scopeElemObjectAttrs.Attributes()["dynamic"].(basetypes.BoolValue).ValueBool()
+			if scopeEntryIsDynamic {
+				if string(scopeEntryName[0]) != "*" {
+					resp.Diagnostics.AddError("Scope name conflict!", fmt.Sprintf("Scope name \"%s\" must be prefixed, suffixed or both by static part(s) with one and only one dynamic part represented by \"*\"", scopeEntryName))
+				}
+			}
+		}
+	}
+
+	// Test exclusive scope names for dynamic true, string must be prepended with *
+	eScopeNames := []string{}
+	if internaltypes.IsDefined(model.ExclusiveScopes) {
+		exclusiveScopeElems := model.ExclusiveScopes.Elements()
+		for _, esElem := range exclusiveScopeElems {
+			esElemObjectAttrs := esElem.(types.Object)
+			eScopeEntryName := strings.ToLower(esElemObjectAttrs.Attributes()["name"].(basetypes.StringValue).ValueString())
+			eScopeNames = append(eScopeNames, eScopeEntryName)
+			eScopeEntryIsDynamic := esElemObjectAttrs.Attributes()["dynamic"].(basetypes.BoolValue).ValueBool()
+			if eScopeEntryIsDynamic {
+				if string(eScopeEntryName[0]) != "*" {
+					resp.Diagnostics.AddError("Dynamic scope name conflict!", fmt.Sprintf("Dynamic scope name \"%s\" must be prefixed, suffixed or both by static part(s) with one and only one dynamic part represented by \"*\"", eScopeEntryName))
+				}
+			}
+		}
+	}
+
+	// Test if values in sets match
+	matchNameBtwnScopes, matchVal := internaltypes.MatchStringInSets(scopeNames, eScopeNames)
+	if matchNameBtwnScopes {
+		resp.Diagnostics.AddError("Scope name conflict!", fmt.Sprintf("The scope name \"%s\" is already defined in another scope list", matchVal))
+	}
+}
+
+func addOptionalOauthAuthServerSettingsFields(ctx context.Context, addRequest *client.AuthorizationServerSettings, plan oauthAuthServerSettingsResourceModel) error {
+
+	if internaltypes.IsDefined(plan.DefaultScopeDescription) {
+		addRequest.DefaultScopeDescription = plan.DefaultScopeDescription.ValueString()
+	}
+
+	if internaltypes.IsDefined(plan.Scopes) {
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.Scopes)), &addRequest.Scopes)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.ScopeGroups) {
+		addRequest.ScopeGroups = client.NewAuthorizationServerSettingsWithDefaults().ScopeGroups
```

Comment:
Why is this needed?

Reply:
This one is fixed but there are similar ones below that start with defaults rather than reading straight from the plan

Reply:
It wasn't 🥇

Reply:
the remaining properties set addRequest to use the type with defaults as properties within the property object are not nullable